server/lib/db/index.js

```diff
@@ -200,6 +200,13 @@ class DatabaseConnector {
       if (data[key] && typeof data[key] === 'string') {
         data[key] = this.bid(data[key]);
       }
+
+      if(data[key] && Array.isArray(data[key])) { 
+        var that = this;
```

Comment:
This looks like a useful case to handle! 

I have only briefly scanned the code here but I think that this can be written as: 
```js 
// map the bid convert method across all elements in the value array
data[key] = data[key].map(this.bid);
```

The way the closure should be respected and it reads exactly as it performs - map this function to these values. 
It would also be useful to have a comment here showing what case this is handling (like the previous `string` case has).